### DISCUSSION TITLE
MS Teams notification using docker-compose

### DISCUSSION THREAD

**Discussion by diogoduartec** [containrrr/watchtower#587] [Q&A] [answered]
**I have following docker-compose structure:**

![image](https://user-images.githubusercontent.com/31852339/86072116-d90dd800-ba4e-11ea-831b-ca7b7ec357d9.png)

**Then I got following error:**
`Failed to send MSTeams notificattion:  Post "https://outlook.office.com....": dial tcp: lookup outlook.office.com on 127.0.0.11:53: read udp 127.0.0.1:48955->127.0.0.11:53: i/o timeout`

**Comment by simonaronsson** [comment]
Hi!

I would suggest that you use the [shoutrrr service](https://containrrr.dev/watchtower/notifications/#containrrrshoutrrr) for that. 

If you insist on using it through the legacy teams notification service, you need to add a URL to your hook URL in the env var collection. You should also remove true from your command, as mentioning a boolean flag is enough to make it true. Currently, you're telling watchtower to monitor the container "provider" and the container "true". 😅

Best,
Simme

**Reply by diogoduartec** [reply]
Actually, `WATCHTOWER_NOTIFICATION_MSTEAMS_HOOK_URL` it's an environment variable. I removed `true` from command, thanks for letting me know.

I tried `shoutrrr` approach, but I'm getting 400 error and I think it's because of `WATCHTOWER_NOTIFICATION_TEMPLATE`

**My docker-compose:**

![image](https://user-images.githubusercontent.com/31852339/86126747-d5a93980-baac-11ea-8845-318b86ad6b81.png)

**My environment variable:**

![image](https://user-images.githubusercontent.com/31852339/86126801-ea85cd00-baac-11ea-9f64-29d85e6264ab.png)


**How I did get token-a, token-b, token-c:**

![image](https://user-images.githubusercontent.com/31852339/86126672-bad6c500-baac-11ea-9b55-ad808b38f29e.png)

**My result:**

`Failed to send notification via shoutrrr (url=teams://...):  failed to send notification to teams, response status code 400 Bad Request`

**Comment by piksel** [comment]
Also, the error message indicates that the DNS lookup (to localhost) failed.

**Reply by diogoduartec** [reply]
What does it mean? `WATCHTOWER_NOTIFICATION_MSTEAMS_HOOK_URL` it's wrong?

**Comment by diogoduartec** [comment, marked as answer]
It worked!

I didn't understand what exactly was wrong, but this is how I made it works:

**My docker-compose:**

![image](https://user-images.githubusercontent.com/31852339/86176025-56d6ef80-baf2-11ea-81f8-5748587ed8cd.png)


**My environment variable:**

![image](https://user-images.githubusercontent.com/31852339/86176237-aae1d400-baf2-11ea-8208-abf8566542ca.png)

**Comment by khentricch** [comment]
Please I have an issue with my termux trying to clone GitHub link for shellphish.
![Screenshot_20200702-091344](https://user-images.githubusercontent.com/67731691/86334641-674baf00-bc45-11ea-984f-00c5fd2614db.png). Please I really need help

**Reply by simonaronsson** [reply]
This is not the right repository for that question. Sorry. 😅

**Comment by T0T4R4** [comment]
Not sure if things have changed on Microsoft side since the last post, but in our case it doesn't work.
We have an enterprise account with O365 and are under SSO, which means our webhook doesn't start the same way as yours :
`https://companyName.webhook.office.com/webhookb2/token-a/IncomingWebhook/token-b/token-c`

When ran once, Watchtower ends with 
'Waiting for the notification goroutine to finish'

and nothing comes up to MsTeams

**Reply by piksel** [reply]
This is due to: https://github.com/containrrr/shoutrrr/issues/138

It's fixed in `latest-dev` (but note that latest-dev is bleeding edge and will not be stable).
